### PR TITLE
[Enhancement] Fast orphan segment deletion of cloud native

### PR DESCRIPTION
**Pull request by sduzh**
During the process of tablet metadata GC, if it is found that a certain metadata to be deleted is generated by compaction task, the input files of that compaction is directly deleted. In this way, the space of persistence storage and local disk cache can be released more immediately.

TODO: Support primary key model.

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by wyb — APPROVED** [review]

File: be/src/storage/lake/gc.cpp [hunk 1]
```diff
@@ -105,6 +105,19 @@ static Status list_tablet_metadata(const std::string& metadir, std::set<std::str
     });
 }
 
+static Status delete_rowset_files(FileSystem* fs, std::string_view data_dir, const RowsetMetadataPB& rowset) {
+    for (const auto& segment : rowset.segments()) {
+        auto seg_path = join_path(data_dir, segment);
+        if (auto st = ignore_not_found(fs->delete_file(seg_path)); st.ok()) {
+            LOG(INFO) << "Deleted " << seg_path;
```
Comment (imay) [inline be/src/storage/lake/gc.cpp:112]: How about deleting this log to reduce log?

Comment (sduzh) [reply be/src/storage/lake/gc.cpp:112]: How about add a configuration to control whether print this log or not? IMHO, file deletion is a quite dangerous operation and once unexpected file deletion happed, it's hard to reason about it without the delete log.

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**